src/Moryx.Resources.Management.Endpoints/ResourceManagementController.cs

```diff
@@ -0,0 +1,375 @@
+
+            return ids.Select(id => _resourceModification.Read(id, r => converter.GetDetails(r)))
+                .Where(details => details != null).ToArray();
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status417ExpectationFailed)]
+        [Route("query")]
+        public ActionResult<ResourceModel[]> GetResources(ResourceQuery query)
+        {
+            var filter = new ResourceQueryFilter(query, _resourceTypeTree);
+            var resourceProxies = _resourceModification.GetResources<PublicResource>(filter.Match).ToArray();
+
+            var converter = new ResourceQueryConverter(_resourceTypeTree, _serialization, query);
+            return resourceProxies.Select(p => _resourceModification.Read(p.Id, r => converter.QueryConversion(r))).ToArray();
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status417ExpectationFailed)]
+        [Route("{id}")]
+        public ActionResult<ResourceModel> GetDetails(long id)
+        {
+            var converter = new ResourceToModelConverter(_resourceTypeTree, _serialization);
+            var resourceModel = _resourceModification.Read(id, r => converter.GetDetails(r));
+            if (resourceModel is null)
+                return NotFound($"Resource '{id}' not found!");
+
+            return resourceModel;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status417ExpectationFailed)]
+        [Route("{id}/invoke/{method}")]
+        public ActionResult<Entry> InvokeMethod(long id, string method, Entry parameters)
+        {
+            if (_resourceModification.GetResource<PublicResource>(id) is null)
+                return NotFound($"Resource '{id}' not found!");
+
+            Entry entry = null;
+            _resourceModification.Modify(id, r =>
+            {
+                entry = EntryConvert.InvokeMethod(r.Descriptor, new MethodEntry { Name = method, Parameters = parameters }, _serialization);
+                return true;
+            });
+            if (entry is null)
+                return Conflict("Method could not be invoked.");
+
+            return entry;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status417ExpectationFailed)]
+        [Route("types/{type}")]
+        public ActionResult<ResourceModel> ConstructWithParameters(string type, string method = null, [FromBody(EmptyBodyBehavior = EmptyBodyBehavior.Allow)] Entry arguments = null)
+        {
+            var trustedType = WebUtility.HtmlEncode(type);
```

Comment:
Resource types use fullname, If you encode them, these become %2E - good luck finding that type